ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/operations/DivOperation.java

```diff
@@ -33,11 +35,14 @@ public Gas cost(final MessageFrame frame) {
 
   @Override
   public void execute(final MessageFrame frame) {
-    final UInt256 value0 = frame.popStackItem().asUInt256();
-    final UInt256 value1 = frame.popStackItem().asUInt256();
-
-    final UInt256 result = value0.dividedBy(value1);
+    final UInt256 value0 = UInt256.fromBytes(frame.popStackItem());
+    final UInt256 value1 = UInt256.fromBytes(frame.popStackItem());
 
-    frame.pushStackItem(result.getBytes());
+    if (value1.isZero()) {
```

Comment:
This optimization is out of scope of this PR in my opinion.

Reply:
As I mention elsewhere, the UInt256 class will throw if you divide by zero.

It's better to handle this here, since this is not actually a proper math behavior.

Reply:
Sounds good.